sdk/src/test/scala/io/horizen/account/state/ContractInteropCallTest.scala

```diff
@@ -1,87 +1,799 @@
 import io.horizen.evm._
 import io.horizen.utils.BytesUtils
-import org.junit.Assert.assertArrayEquals
+import org.junit.Assert.{assertArrayEquals, assertEquals, assertTrue, fail}
 import org.junit.Test
 import sparkz.crypto.hash.Keccak256
 
 import java.math.BigInteger
+import java.nio.charset.StandardCharsets
+import scala.util.{Failure, Try}
 
 class ContractInteropCallTest extends ContractInteropTestBase {
   override val processorToTest: NativeSmartContractMsgProcessor = NativeTestContract
 
+
+  val WRITE_PROTECTION_ERR_MSG_FROM_EVM = "write protection"
+  val WRITE_PROTECTION_ERR_MSG_FROM_NATIVE_CONTRACT = "invalid write access to storage"
+
   private object NativeTestContract extends NativeSmartContractMsgProcessor {
     override val contractAddress: Address = new Address("0x00000000000000000000000000000000deadbeef")
     override val contractCode: Array[Byte] = Keccak256.hash("NativeTestContract")
 
+    val STATICCALL_READONLY_TEST_SIG = "aaaaaaaa"
+    val STATICCALL_READWRITE_TEST_SIG = "bbbbbbbb"
+    val STATICCALL_READWRITE_WITH_TRY_TEST_SIG = "cccccccc"
+    val STATICCALL_NESTED_CALLS_TEST_SIG = "dddddddd"
+    val CREATE_TEST_SIG = "eeeeeeee"
+
+    val COUNTER_KEY = Keccak256.hash("key".getBytes(StandardCharsets.UTF_8))
+
+    val SUB_CALLS_GAS: BigInteger =  BigInteger.valueOf(25000)
+    val SUB_CALLS_GAS_HEX_STRING =  "0x" + SUB_CALLS_GAS.toString(16)
+
+    val DEPLOY_GAS: BigInteger = BigInteger.valueOf(320000)
+    val DEPLOY_GAS_HEX_STRING =  "0x" + DEPLOY_GAS.toString(16)
+
+
     override def process(
         invocation: Invocation,
         view: BaseAccountStateView,
         context: ExecutionContext
     ): Array[Byte] = {
-      // read target contract address from input
-      val evmContractAddress = new Address(invocation.input)
-      // function signature of retrieve()
-      val retrieveSignature = BytesUtils.fromHexString("2e64cec1")
+
+      val gasView = view.getGasTrackedView(invocation.gasPool)
+      //read method signature
+     getFunctionSignature(invocation.input) match {
+        case STATICCALL_READONLY_TEST_SIG => testStaticCallOnReadonlyMethod(invocation, gasView, context)
+        case NATIVE_CONTRACT_RETRIEVE_ABI_ID => retrieve(gasView)
+        case STATICCALL_READWRITE_TEST_SIG => testStaticCallOnReadwriteMethod(invocation, gasView, context)
+        case NATIVE_CONTRACT_INC_ABI_ID => inc(gasView)
+        case STATICCALL_READWRITE_WITH_TRY_TEST_SIG => testStaticCallOnReadwriteMethodWithTry(invocation, gasView, context)
+        case STATICCALL_NESTED_CALLS_TEST_SIG => testStaticCallNestedCalls(invocation, gasView, context)
+        case CREATE_TEST_SIG => testDeployContract(invocation, gasView, context)
+        case _ => throw new IllegalArgumentException("Unknown method call")
+      }
+    }
+
+    def testStaticCallOnReadonlyMethod(
+                                  invocation: Invocation,
+                                  view: BaseAccountStateView,
```

Comment:
True, in fact I left the same signature of `process`  method just in case in the future the parameter may be used, but in the end it is better to remove it.

Done